src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -136,49 +137,47 @@ public BitArray(bool[] values)
             ref byte value = ref Unsafe.As<bool, byte>(ref MemoryMarshal.GetArrayDataReference<bool>(values));
             if (Vector512.IsHardwareAccelerated)
             {
-                for (; i <= (uint)values.Length - Vector512<byte>.Count; i += (uint)Vector512<byte>.Count)
+                for (; i <= values.Length - Vector512<byte>.Count; i += Vector512<byte>.Count)
                 {
-                    Vector512<byte> vector = Vector512.LoadUnsafe(ref value, i);
+                    Vector512<byte> vector = Vector512.LoadUnsafe(ref value, (uint)i);
                     Vector512<byte> isFalse = Vector512.Equals(vector, Vector512<byte>.Zero);
 
-                    ulong result = isFalse.ExtractMostSignificantBits();
-                    m_array[i / 32u] = (int)(~result & 0x00000000FFFFFFFF);
-                    m_array[(i / 32u) + 1] = (int)((~result >> 32) & 0x00000000FFFFFFFF);
+                    ulong result = ~(isFalse.ExtractMostSignificantBits());
+                    m_array[i / 32] = (int)result;
+                    m_array[(i / 32) + 1] = (int)(result >> 32);
                 }
             }
-            else if (Vector256.IsHardwareAccelerated)
+            if (Vector256.IsHardwareAccelerated)
             {
-                for (; i <= (uint)values.Length - Vector256<byte>.Count; i += (uint)Vector256<byte>.Count)
+                for (; i <= values.Length - Vector256<byte>.Count; i += Vector256<byte>.Count)
                 {
-                    Vector256<byte> vector = Vector256.LoadUnsafe(ref value, i);
+                    Vector256<byte> vector = Vector256.LoadUnsafe(ref value, (uint)i);
                     Vector256<byte> isFalse = Vector256.Equals(vector, Vector256<byte>.Zero);
 
-                    uint result = isFalse.ExtractMostSignificantBits();
-                    m_array[i / 32u] = (int)(~result);
+                    m_array[i / 32] = (int)~(isFalse.ExtractMostSignificantBits());
                 }
             }
             else if (Vector128.IsHardwareAccelerated)
```

Comment:
All the other similar code dropped the `else` for this branch.

Reply:
Here, the loop conditions for `Vector256` and `Vector128` paths are basically same (`i <= values.Length - 32`), so there is no point executing the `Vector128` path if the `Vector256` path is already executed.